sharktank/tests/models/llama/benchmark-tests.py

```diff
@@ -0,0 +1,310 @@
+            f"--iree-hal-dump-executable-files-to={self.repo_root}/files/llama",
+        ]
+        self.prefill_args_f16 = artifacts_dir + "prefill_args"
+        self.decode_args_f16 = artifacts_dir + "decode_args"
+        self.prefill_args_fp8 = artifacts_dir + "prefill_args_fp8"
+        self.decode_args_fp8 = artifacts_dir + "decode_args_fp8"
+        self.iree_run_prefill_args = [
+            "--function=prefill_bs4",
+            f"--input=@{self.prefill_args_f16}/tokens.npy",
+            f"--input=@{self.prefill_args_f16}/seq_lens.npy",
+            f"--input=@{self.prefill_args_f16}/seq_block_ids.npy",
+            f"--input=@{self.prefill_args_f16}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+        self.iree_run_decode_args = [
+            "--function=decode_bs4",
+            f"--input=@{self.decode_args_f16}/tokens.npy",
+            f"--input=@{self.decode_args_f16}/seq_lens.npy",
+            f"--input=@{self.decode_args_f16}/start_positions.npy",
+            f"--input=@{self.decode_args_f16}/seq_block_ids.npy",
+            f"--input=@{self.decode_args_f16}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+        self.iree_run_prefill_args_fp8 = [
+            "--function=prefill_bs4",
+            f"--input=@{self.prefill_args_fp8}/tokens.npy",
+            f"--input=@{self.prefill_args_fp8}/seq_lens.npy",
+            f"--input=@{self.prefill_args_fp8}/seq_block_ids.npy",
+            f"--input=@{self.prefill_args_fp8}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+        self.iree_run_decode_args_fp8 = [
+            "--function=decode_bs4",
+            f"--input=@{self.decode_args_fp8}/tokens.npy",
+            f"--input=@{self.decode_args_fp8}/seq_lens.npy",
+            f"--input=@{self.decode_args_fp8}/start_positions.npy",
+            f"--input=@{self.decode_args_fp8}/seq_block_ids.npy",
+            f"--input=@{self.decode_args_fp8}/cache_state_f16.npy",
+            "--benchmark_repetitions=3",
+        ]
+
+    def testBenchmark8B_f16_Decomposed(self):
+        self.export_mlir(
+            "decomposed",
+            self.irpa_path,
+            self.output_mlir,
+            self.output_json,
+            self.repo_root,
+        )
+        self.iree_compile(
+            self.output_mlir, self.output_vmfb, self.iree_compile_args, self.repo_root
+        )
+        # benchmark prefill
+        self.iree_benchmark_module(
+            "0",
+            self.output_vmfb_fp8,
+            self.irpa_path_fp8,
+            self.iree_run_prefill_args_fp8,
+            self.repo_root,
+        )
+        # benchmark decode
+        self.iree_benchmark_module(
+            "0",
+            self.output_vmfb_fp8,
+            self.irpa_path_fp8,
+            self.iree_run_decode_args_fp8,
+            self.repo_root,
+        )
+
+    @pytest.mark.xfail
+    def testBenchmark8B_f16_Non_Decomposed(self):
```

Comment:
Instead of marking the whole test as xfail, can we xfail the particular step it fails on? That way we can track as we make progress or regress with certain configs

Reply:
I can do this in a follow-up PR.

Reply:
Done